### PR TITLE
cilium: fix encryption flow labels in ip6 case

### PR DESCRIPTION
**Pull request by jrfastab**
We don't want to attach ip6 flow labels and set mark fields if the packet is only going to the stack to be encrypted. So short circuit the pass_to_stack logic after we decide the packet needs encryption and send directly to stack.

Also fix up DatapathConfiguration tests so they run on my local cluster and are in-sync with tests that do not use encryption.

### CODE REVIEW

**Comment by jrfastab** [comment]
test-focus K8sDatapathConfig.transparent

File: bpf/bpf_lxc.c [hunk 1]
```diff
@@ -663,7 +663,7 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx,
 		 * for further processing.
 		 */
 		else if (ret == IPSEC_ENDPOINT)
-			goto pass_to_stack;
+			return CTX_ACT_OK;
```
Comment (joestringer) [inline bpf/bpf_lxc.c]: We lose the monitor messages with this approach, is that intended?

Comment (jrfastab) [reply bpf/bpf_lxc.c]: It was intended but agree its not such a good idea. I'll move it around so we don't lose this.

**Comment by jrfastab** [comment]
test-focus K8sDatapathConfig

**Comment by jrfastab** [comment]
Doing a preliminary test on datapath configurations I have what I believe is an unrelated issue with my local cluster but above runs in gke cluster. The patch is good I believe and the local cluster issue seems to be related to iptables/route/mtu conflicts somehow. I get spurious TCP reset both with above patch and before.

**Comment by jrfastab** [comment]
build error when encap is not enabled :( but on the plus side transparent encryption + vxlan passed. I'll push an update and rerun.

**Comment by jrfastab** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

**Comment by aanm** [comment]
hit https://github.com/cilium/cilium/issues/10929

**Comment by aanm** [comment]
hit https://github.com/cilium/cilium/issues/10929

**Pull request merged**